File: leetcode/hash tables/1. Two Sum/python/main.py
# from ast import List
# class Solution:
#     def twoSum(self, nums: List[int], target: int) -> List[int]:
#         ans = []
#         for i in range(len(nums) - 1):
#             for j in range(i + 1, len(nums)):
#                 if nums[i] + nums[j] == target:
#                     ans.extend([i, j])
#         return ans
# nums = [3, 2, 3]
# target = 6
# print(Solution().twoSum(nums, target))


from ast import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        o = {}
        for i in range(len(nums)):
            d = target - nums[i]
            if d in o:
                return [i, o[d]]
            o[nums[i]] = i


nums = [3, 2, 3]
target = 6
print(Solution().twoSum(nums, target))

File: leetcode/hash tables/1. Two Sum/golang/main.go
package main

import "fmt"

func twoSum(nums []int, target int) []int {
	var ans []int
	o := map[int]int{}
	for i, v := range nums {
		d := target - v
		if idx, ok := o[d]; ok {
			ans = []int{i, idx}
		}
		o[v] = i
	}
	return ans
}
func main() {
	nums := []int{2, 7, 11, 15}
	target := 9
	fmt.Println(twoSum(nums, target))
}
